packages/cloudflare/src/api/kvCache.ts

```diff
@@ -62,6 +79,17 @@ class Cache implements IncrementalCache {
           };
         }
       }
+
+      const entryValue = entry?.value as CachedFetchValue | undefined;
+      if (entryValue?.kind === "FETCH") {
```

Comment:
Don't we only want to do this when the value comes from the assets?

Reply:
why wouldn't we want to check the expiration date also for entries from the KV?

Reply:
My understanding is that this is done by the caller.

My understanding of the issue: when an entry is expired, Next will use the [stalled entry](https://nextjs.org/docs/app/building-your-application/caching#time-based-revalidation) and update the cache in the background.

If there is no KV then we can not update the cache and we will keep serving the stalled entry.

Then I think that we don't want to use the stalled entry in this case only.

So I think my previous comment should be

```diff
-Don't we only want to do this when the value comes from the assets?
+Don't we only want to do this when there is no KV?
```